### PR TITLE
Temporal: Move ZonedDateTime order-of-operations tests out of staging

### PR DESCRIPTION
**Pull request by catamorphism**

### CODE REVIEW

**Review by ptomato — APPROVED** [review]
Nitpick about names: to avoid confusion with the existing `order-of-operations.js` tests which test order of observable user calls, let's rename these to something like `math-order-of-operations-*.js`. Also, the "reject" ones aren't actually testing any order of operations, so let's rename those to `overflow-reject.js`.

Otherwise looks good.

**Comment by ptomato** [comment]
Oh, we also should expand these tests to cover PlainDate and PlainDateTime arithmetic, but I'll add that to the list of followups.

File: tools/scripts/ci_lint.sh [hunk 1]
```diff
@@ -15,4 +15,4 @@ else
   paths="test/"
 fi
 
-./tools/lint/lint.py --exceptions lint.exceptions $paths
+python3 ./tools/lint/lint.py --exceptions lint.exceptions $paths
```
[the remark this answers is not in the document]
Comment (catamorphism) [reply tools/scripts/ci_lint.sh:18]: Oops, yes -- fixed.

**Review by ptomato — APPROVED** [review]
LGTM!

**Pull request merged**